weather.gismeteo/resources/libs/simpleweather.py

```diff
@@ -0,0 +1,725 @@
+                '07' : 27,
+                '08' : 28,
+                '09' : 29,
+                '10' : 30,
+                '11' : 31,
+                '12' : 32,
+                }
+
+    @staticmethod
+    def month_name_short():
+
+        return {
+                '01' : 51,
+                '02' : 52,
+                '03' : 53,
+                '04' : 54,
+                '05' : 55,
+                '06' : 56,
+                '07' : 57,
+                '08' : 58,
+                '09' : 59,
+                '10' : 60,
+                '11' : 61,
+                '12' : 62,
+                }
+
+    @staticmethod
+    def week_day_long():
+
+        return {
+                '0' : 17,
+                '1' : 11,
+                '2' : 12,
+                '3' : 13,
+                '4' : 14,
+                '5' : 15,
+                '6' : 16,
+                }
+
+    @staticmethod
+    def week_day_short():
+
+        return {
+                '0' : 47,
+                '1' : 41,
+                '2' : 42,
+                '3' : 43,
+                '4' : 44,
+                '5' : 45,
+                '6' : 46,
+                }
+
+
+class Dialogs():
+
+    def notify_error(self, error, show_dialog=False):
+        heading = ''
+        message = '{0}'.format(error)
+        if isinstance(error, WebClientError):
+            heading = self.gettext('Connection error')
+        else:
+            self.log_error(message)
+
+        if show_dialog:
+            if heading:
+                self.dialog_ok(heading, message)
+            else:
+                self.dialog_ok(message)
+        else:
+            self.dialog_notification_error(heading, message)
+
+    def dialog_notification_error(self, heading, message="", time=0, sound=True):
+        self.dialog_notification(heading, message, xbmcgui.NOTIFICATION_ERROR, time, sound)
+
+    def dialog_notification_info(self, heading, message="", time=0, sound=True):
+        self.dialog_notification(heading, message, xbmcgui.NOTIFICATION_INFO, time, sound)
+
+    def dialog_notification_warning(self, heading, message="", time=0, sound=True):
+        self.dialog_notification(heading, message, xbmcgui.NOTIFICATION_WARNING, time, sound)
+
+    def dialog_notification(self, heading, message="", icon="", time=0, sound=True):
+
+        _message = message if message else heading
+
+        if heading \
+          and heading != _message:
+            _heading = '{0}: {1}'.format(self.name, heading)
+        else:
+            _heading = self.name
+
+        xbmcgui.Dialog().notification(_heading, _message, icon, time, sound)
+
+    def dialog_ok(self, line1, line2="", line3=""):
+
+        if self.kodi_major_version() >= '19':
+            xbmcgui.Dialog().ok(self.name, self._join_strings(line1, line2, line3))
+        else:
+            xbmcgui.Dialog().ok(self.name, line1, line2, line3)
+
+    def dialog_progress_create(self, heading, line1="", line2="", line3=""):
```

Comment:
As a small reminder: the line1, line2, line3 will be removed in Matrix: https://forum.kodi.tv/showthread.php?tid=344263

Reply:
For Matrix and newer, I use `self._join_strings()` to join these lines into one